app/controllers/sessions_controller.rb

```diff
@@ -170,4 +172,13 @@ def terms_accepted?(user, community)
     user && community.consent.eql?(user.consent)
   end
 
+  def get_origin_locale
```

Comment:
This should take request as input. The style to follow is to restrict access to top level globals to controller actions but have them passed to helper methods explicitly. This helps with both readability and helper method reusability/refactorability.